5-AccessControl/2-call-api-groups/README.md

```diff
@@ -85,161 +105,182 @@ For more information and potential issues, see: [HTTPS in .NET Core](https://doc
+ 
+##### Publish Delegated Permissions
+
+1. All APIs must publish a minimum of one [scope](https://docs.microsoft.com/azure/active-directory/develop/v2-oauth2-auth-code-flow#request-an-authorization-code), also called [Delegated Permission](https://docs.microsoft.com/azure/active-directory/develop/v2-permissions-and-consent#permission-types), for the client apps to obtain an access token for a *user* successfully. To publish a scope, follow these steps:
+1. Select **Add a scope** button open the **Add a scope** screen and Enter the values as indicated below:
+    1. For **Scope name**, use `access_via_group_assignment`.
+    1. Select **Admins and users** options for **Who can consent?**.
+    1. For **Admin consent display name** type in *Access 'msal-angular-app' as the signed-in user.*.
+    1. For **Admin consent description** type in *Allow the app to access the 'msal-angular-app' as a signed-in user.*.
+    1. For **User consent display name** type in *Access 'msal-angular-app' on your behalf.*.
+    1. For **User consent description** type in *Allow the app to access the 'msal-angular-app' on your behalf.*.
+    1. Keep **State** as **Enabled**.
+    1. Select the **Add scope** button on the bottom to save this scope.
+1. Select the **Manifest** blade on the left.
+    1. Set `accessTokenAcceptedVersion` property to **2**.
+    1. Select on **Save**.
+
+> :information_source:  Follow  [the principle of least privilege](https://docs.microsoft.com/azure/active-directory/develop/secure-least-privileged-access) whenever you are publishing permissions for a web API.
+
+##### Grant Delegated Permissions to msal-angular-app
+
+1. Since this app signs-in users, we will now proceed to select **delegated permissions**, which is is required by apps signing-in users.
+   1. In the app's registration screen, select the **API permissions** blade in the left to open the page where we add access to the APIs that your application needs:
+   1. Select the **Add a permission** button and then:
+   1. Ensure that the **My APIs** tab is selected.
+   1. In the list of APIs, select the API `msal-angular-app`.
+        1. Since this app signs-in users, we will now proceed to select **delegated permissions**, which is is requested by apps when signing-in users.
+            1. In the **Delegated permissions** section, select the **Access 'msal-angular-app'** in the list. Use the search box if necessary.
+   1. Select the **Add permissions** button at the bottom.
+   1. Select the **Add a permission** button and then:
+   1. Ensure that the **Microsoft APIs** tab is selected.
+   1. In the *Commonly used Microsoft APIs* section, select **Microsoft Graph**
+      * Since this app signs-in users, we will now proceed to select **delegated permissions**, which is is requested by apps when signing-in users.
+           1. In the **Delegated permissions** section, select the **User.Read**, **GroupMember.Read.All** in the list. Use the search box if necessary.
```

Comment:
>**GroupMember.Read.All** [](http://example.com/codeflow?start=82&length=24)

We need a way to place the text asking for admin consent here, maybe a new non-mandatory flag named RequiresAdminConsent on the app object itself?

Reply:
Good idea. Let me add it in the codegen

Reply:
https://github.com/Azure-Samples/GenerateScriptsForSample/issues/44